docs/groovy-script/getting_started/reloading.md

```diff
@@ -0,0 +1,77 @@
+---
+order: 1000
+---
+
+# Reloading
+
+
+GroovyScript recipes can be reloaded without having to do a full restart.
+Any script file using the `postInit` loader can be reloaded.
+
+::: info Danger {id="danger"}
+
+Some compat may not properly implement reloading, and will not be reloaded properly or may have errors while reloading.
+
+:::
+
+To reload GroovyScript, either:
+
+- Run the `/gs reload` command
+- Use the keybind, by default bound to `CTRL+R`
+
+
+## Class Caching
+
+For performance reasons, script classes are cached.
+In some situations, they may not be properly reloaded in-game.
+You can delete the cache to force these classes to be reloaded.
+
+- Run the `/gs deleteScriptCache` command and then reload normally
+
+
+## Add Custom Reloading
+
+For reloading to be added, each compat needs to be explicit handled by GroovyScript.
+
+However, you can add your own custom class and implement reloading for your recipes even if GroovyScript doesn't have support for the given recipes natively.
+To do so, first make a new file in `classes` with the class containing an `add`, `remove`, and an `onReload` method.
+It should have an instance created inside it for access purposes.
```

Comment:
i had modified something that worked in Java, and presumed it would work in Groovy. i intended to test it before committing it, but forgot to do so.
i have now tested it, and this does not work, but primarily due to the blacklisting of GroovyScript packages. after editing it a bit, i have managed to get it to work, and have updated the file accordingly.

on the "do we really want to encourage people":
1. yeah i do, and honestly i dont think its *that* cursed. its just a Groovy version of the `VirtualizedRegistry`.
2. if it functionally allows compat to be created by a modpack developer, thats a significant improvement in the usability of GroovyScript.
3. people are going to try to do it anyways. having a "heres how" just means they dont have to reinvent the wheel.